### PR TITLE
kernel: bump 5.4 to 5.4.61

### PR DESCRIPTION
**Pull request by graysky2**
Manual merged:
backport-5.4
  `011-kbuild-export-SUBARCH.patch`
layerscape
  `701-net-0262-net-dsa-ocelot-add-tagger-for-Ocelot-Felix-switches.patch`

All other modifications made by `update_kernel.sh`

Build-tested: x86_64, lantiq, rampips
Run-tested: ipq806x (R7800), lantiq (xrx200), x86_64, ramips (RT-AC57U)

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

### CODE REVIEW

File: target/linux/generic/backport-5.4/011-kbuild-export-SUBARCH.patch [hunk 1]
```diff
@@ -10,15 +10,15 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
...
  CLANG_FLAGS :=
  
--export ARCH SRCARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE AS LD CC
--export CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE PAHOLE LEX YACC AWK INSTALLKERNEL
-+export ARCH SRCARCH SUBARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE AS LD
-+export CC CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE PAHOLE LEX YACC AWK INSTALLKERNEL
+-export ARCH SRCARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE LD CC
+-export CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE READELF PAHOLE LEX YACC AWK INSTALLKERNEL
++export ARCH SRCARCH SUBARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE AS LD CC
++export CC CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE READELF PAHOLE LEX YACC AWK INSTALLKERNEL
```
Comment (adschm) [inline target/linux/generic/backport-5.4/011-kbuild-export-SUBARCH.patch:20]: If you fold back the CC to the first line here, the second lines is unchanged again, so one could reduce the impact of the patch:

--- a/target/linux/generic/backport-5.4/011-kbuild-export-SUBARCH.patch
+++ b/target/linux/generic/backport-5.4/011-kbuild-export-SUBARCH.patch
@@ -10,15 +10,12 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>

```
 --- a/Makefile
 +++ b/Makefile
-@@ -471,8 +471,9 @@ KBUILD_LDFLAGS :=
+@@ -487,7 +487,7 @@ KBUILD_LDFLAGS :=
  GCC_PLUGINS_CFLAGS :=
  CLANG_FLAGS :=

--export ARCH SRCARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE AS LD CC
--export CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE PAHOLE LEX YACC AWK INSTALLKERNEL
-+export ARCH SRCARCH SUBARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE AS LD
-+export CC CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE PAHOLE LEX YACC AWK INSTALLKERNEL
-+
- export PERL PYTHON PYTHON2 PYTHON3 CHECK CHECKFLAGS MAKE UTS_MACHINE HOSTCXX
+-export ARCH SRCARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE LD CC
++export ARCH SRCARCH SUBARCH CONFIG_SHELL BASH HOSTCC KBUILD_HOSTCFLAGS CROSS_COMPILE LD CC
+ export CPP AR NM STRIP OBJCOPY OBJDUMP OBJSIZE READELF PAHOLE LEX YACC AWK INSTALLKERNEL
+ export PERL PYTHON PYTHON3 CHECK CHECKFLAGS MAKE UTS_MACHINE HOSTCXX
  export KBUILD_HOSTCXXFLAGS KBUILD_HOSTLDFLAGS KBUILD_HOSTLDLIBS LDFLAGS_MODULE
-
```

Comment (graysky2) [reply target/linux/generic/backport-5.4/011-kbuild-export-SUBARCH.patch:20]: I will make the change and rebase.  I did it as such to conform as closely as possible to the original patch.

**Comment by graysky2** [comment]
I couldn't apply the patch you posted for some reason, so I modified myself.  Please review.  Built fine.

**Comment by adschm** [comment]
> I couldn't apply the patch you posted for some reason, so I modified myself. Please review. Built fine. I cannot flash the resulting build image due to network usage here.

Looks fine. You could remove the empty line at the end of the modified patch ...

**Comment by graysky2** [comment]
@adschm - Done.  Rebased.  It builds but again, I cannot test the resulting image due to use of my network.

**Comment by zuzia-dev** [comment]
Build system: Lantiq xrx200 - OK. Everything works.

**Comment by graysky2** [comment]
@adschm - I was able to confirm that the latest rebased commit (https://github.com/openwrt/openwrt/pull/3349/commits/1d526e66d84b9c2f8522065185bf77a940d1918d) builds and gave a functional image on ipq806x.

**Comment by curtdept** [comment]
Build and Run Tested by: Curtis Deptuck <curtdept@me.com> (x86/64)

**Comment by Borromini** [comment]
Built and run-tested on ramips/mt7621 (Asus RT-AC57U) here.

**Comment by graysky2** [comment]
@adschm - Updated commit message to reflect build and run testing of users.

**Comment by adschm** [comment]
Thanks, merged.

**Pull request closed without merging**